tvm/src/codegen/build_util.cc

```diff
@@ -726,76 +535,95 @@ void GenHostCode(TVMArgs& args,
-  for (size_t k = args.size(); k < arg_info.size(); k++) {
-    auto type = std::get<2>(arg_info[k]);
-    auto shape = std::get<3>(arg_info[k]);
-    PrintIndent(stream, indent);
-    stream << Type2Byte(Type2TVMType(type)) << " " << "name[";
-    if (shape.size() > 0) {
-      for (size_t i = 0; i < shape.size(); i++) {
-        if (i != shape.size() - 1)
-          stream << shape[i] << " * ";
-        else stream << shape[i];
-      }
-    } else {
-      stream << "1";
-    }
-    stream << "];\n";
+  if (platform == "sdaccel" || platform == "vitis") {
+    stream << R"(
+  if (argc != 2) {
+      std::cout << "Usage: " << argv[0] << " <XCLBIN File>" << std::endl;
+      return EXIT_FAILURE;
   }
 
-  // generate host side (before kernel)
-  PrintIndent(stream, indent);
-  stream << "printf(\"Finished setting up shared memory\\n\");\n";
-  PrintIndent(stream, indent);
-  stream << "// compute bofore kernel function\n";
-  size_t pos = host_code.find("top(");
-  std::string pre_kernel  = host_code.substr(0, pos -1);
-  std::string post_kernel = host_code.substr(host_code.find('\n', pos) + 1);
-  pre_kernel = pre_kernel.substr(pre_kernel.find_first_not_of("\n"));
-  pre_kernel = pre_kernel.substr(pre_kernel.find_first_not_of(" "));
-  PrintIndent(stream, indent);
-  
-  if (platform == "sdaccel") {
-    // create variable wrapper
-    stream << pre_kernel << "\n";
-    KernelInit(stream, platform, args,
-               arg_types, arg_info);
-  } else if (platform == "vivado_hls") {
-    // init hls stream channels 
-    for (size_t k = 0; k < arg_info.size(); k++) {
-      auto info = arg_info[k]; 
-      if (std::get<1>(info)) {
-        PrintIndent(stream, indent);
-        stream << "hls::stream<" 
-               << PrintHalideType(std::get<2>(info)) 
-               << "> " << "fd_" << std::get<0>(info) << ";\n";
-      }  
-    }
-    PrintIndent(stream, indent);
-    stream << pre_kernel << "\n";
-    PrintIndent(stream, indent);
-    // create kernel call from host 
-    stream << "top(";
-    for (size_t i = 0; i < arg_info.size(); i++) {
-      auto info = arg_info[i];
-      auto name = std::get<0>(info);
-      if (i != 0) stream << ", ";
-      stream << "fd_" << name;
-    }
-    stream << ");\n";
+  auto binaryFile = argv[1];
+  cl_int err = CL_SUCCESS;
+
+  // create binary file and program
+  auto devices = xcl::get_xil_devices();
+  auto device_count = devices.size();
+  auto device = devices[0];
+
+  cl::Context context(device, NULL, NULL, NULL, &err);
+  cl::CommandQueue q(context, device, CL_QUEUE_PROFILING_ENABLE, &err);
+  auto device_name = device.getInfo<CL_DEVICE_NAME>();
+  std::cout << "Found Device=" << device_name.c_str() << std::endl;
+
+  auto fileBuf = xcl::read_binary_file(binaryFile);
+  cl::Program::Binaries bins{{fileBuf.data(), fileBuf.size()}};
+  devices.resize(1);
+  cl::Program program(context, devices, bins, NULL, &err);
+
+)";
+
+  } else if (platform == "aocl") {
+    stream << R"(
```

Comment:
We should move this kind of strings to a file in the future. No need to do it right now.